phys/module_sf_bep.F

```diff
@@ -180,16 +175,17 @@ subroutine BEP(FRC_URB2D,UTYPE_URB2D,itimestep,dz8w,dt,u_phy,v_phy,      &
       integer nzurban(nurbm)                  
 
 !    Building parameters      
-      real alag_u(nurbm)                      ! Ground thermal diffusivity [m^2 s^-1]
-      real alaw_u(nurbm)                      ! Wall thermal diffusivity [m^2 s^-1]
-      real alar_u(nurbm)                      ! Roof thermal diffusivity [m^2 s^-1]
-      real csg_u(nurbm)                       ! Specific heat of the ground material [J m^3 K^-1]
-      real csw_u(nurbm)                       ! Specific heat of the wall material [J m^3 K^-1]
-      real csr_u(nurbm)                       ! Specific heat of the roof material [J m^3 K^-1]
-      real twini_u(nurbm)                     ! Initial temperature inside the building's wall [K]
-      real trini_u(nurbm)                     ! Initial temperature inside the building's roof [K]
-      real tgini_u(nurbm)                     ! Initial road temperature
 !
+      real,allocatable,dimension(:) :: alag_u                      ! Ground thermal diffusivity [m^2 s^-1]
+      real,allocatable,dimension(:) :: alaw_u                      ! Wall thermal diffusivity [m^2 s^-1]
+      real,allocatable,dimension(:) :: alar_u                      ! Roof thermal diffusivity [m^2 s^-1]
+      real,allocatable,dimension(:) :: csg_u                       ! Specific heat of the ground material [J m^3 K^-1]
+      real,allocatable,dimension(:) :: csw_u                       ! Specific heat of the wall material [J m^3 K^-1]
+      real,allocatable,dimension(:) :: csr_u                       ! Specific heat of the roof material [J m^3 K^-1]
+      real,allocatable,dimension(:) :: twini_u                     ! Initial temperature inside the building's wall [K]
+      real,allocatable,dimension(:) :: trini_u                     ! Initial temperature inside the building's roof [K]
+      real,allocatable,dimension(:) :: tgini_u                     ! Initial road temperature
+
```

Comment:
Since you already pass the nurbm to BEP subroutine, why do you need to modify the declaration here? The nurbm will have a value (3 or 11) when you declare/define these variable dimensions here, right? This comment is also for similar changes below.

Reply:
For these 3 variable probably this declaration is useless